### PR TITLE
Include invocation start times

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
For multi-invocation builders (e.g., dist-x86_64-linux) this timestamp is necessary to correlate the data in the metrics JSON with other data sources (e.g., logs, cpu-usage CSV, etc.). Such correlation may not be perfect but is sometimes helpful and awkward to do otherwise.

### CODE REVIEW

File: src/bootstrap/metrics.rs [hunk 1]
```diff
@@ -124,6 +125,11 @@ impl BuildMetrics {
             }
         };
         invocations.push(JsonInvocation {
+            start_time: state
+                .invocation_start
+                .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                .unwrap_or(Duration::ZERO)
```
Comment (ozkanonur) [inline src/bootstrap/metrics.rs:131]: Will `Duration::ZERO` ever be useful? Why not panicking directly?

Comment (Mark-Simulacrum) [reply src/bootstrap/metrics.rs]: Yeah, panicking seems fine. I don't think we should ever cross the epoch boundary in CI.

**Comment by ozkanonur** [comment]
r=me once the CI is green

**Pull request merged**